### PR TITLE
Implemented Vector

### PR DESCRIPTION
**Pull request by CinchBlue**
I have implemented **Vector** which is similar to [Bytevector](https://github.com/uhmanoa-transpiler-project/shaka-scheme/blob/master/src/shaka_scheme/system/base/Bytevector.hpp).

**Please check whether if there are any memory leaks regarding new/delete, copy/move constructors, and destructors.**

### CODE REVIEW

File: src/shaka_scheme/system/base/Vector.cpp
Comment (btwooton) [inline src/shaka_scheme/system/base/Vector.cpp:54]: The alternative approach is to say `arr(nullptr), size(0) { swap(*this, other); } ` However, your approach works too, although I do not believe the calls to `std::move` on `other.arr` and `other.size` are necessary, since arr is just a raw pointer, and size is just an unsigned number. `std::move` just casts these things to R values, but I am not aware that move semantics really apply to raw pointers, or numeric primitive types in general that is.

**Review by btwooton — APPROVED** [review]
Looks good to me. I don't see anything glaringly obviously wrong with any of your constructors, or with your copy/move semantics. Your tests appear to cover all of the bases. I might suggest having either @1010jms or @seanytak do a once over prior to final approval and merging, just to be safe and avoid any oversights, but I'll leave that up to you.

File: src/shaka_scheme/system/base/Vector.hpp
Comment (btwooton) [inline src/shaka_scheme/system/base/Vector.hpp:13]: Is there a specific reason that you preferred a raw pointer to a dynamically allocated array of NodePtr instead of an `std::vector<NodePtr>`? I'm mostly just curious.

Comment (CinchBlue) [reply src/shaka_scheme/system/base/Vector.hpp:13]: Yes -- `shaka::Vector` is always going to be a fixed size -- it cannot be appended to and so any functions that would "make it bigger" would simply return a bigger `shaka::Vector` and therefore reallocate anyways.

**Pull request merged**
